picard/ui/scripteditor.py

```diff
@@ -0,0 +1,547 @@
+        """
+        self.synchronize_selected_example_lines(self.current_row, self.ui.example_filename_after, self.ui.example_filename_before)
+
+    def save_script(self):
+        """Emits a `save` signal to trigger appropriate save action in the parent object.
+        """
+        self.signal_save.emit()
+
+    def get_script(self):
+        """Provides the text of the file naming script currently loaded into the editor.
+
+        Returns:
+            str: File naming script
+        """
+        return str(self.ui.file_naming_format.toPlainText()).strip()
+
+    def set_script(self, script_text):
+        """Sets the text of the file naming script into the editor.  Sets default text if `script_text` is empty or missing.
+
+        Args:
+            script_text (str): File naming script text to set in the editor.
+        """
+        self.ui.file_naming_format.setPlainText(str(script_text).strip())
+
+    def update_example_files(self):
+        """Update the before and after file naming examples list.
+        """
+        self.examples.update_sample_example_files()
+        self.display_examples()
+
+    def update_examples(self):
+        """Update the before and after file naming examples using the current file naming script in the editor.
+        """
+        override = {'file_naming_format': self.get_script()}
+        self.examples.update_examples(override)
+        self.display_examples()
+
+    @staticmethod
+    def update_example_listboxes(before_listbox, after_listbox, examples):
+        before_listbox.clear()
+        after_listbox.clear()
+        for before, after in sorted(examples, key=lambda x: x[1]):
+            before_listbox.addItem(before)
+            after_listbox.addItem(after)
+
+    def display_examples(self, send_signal=True):
+        """Update the display of the before and after file naming examples.  Optionally emits an `update` signal.
+
+        Args:
+            send_signal (bool, optional): Determines if an `update` signal is emitted. Defaults to True.
+        """
+        self.current_row = -1
+        examples = self.examples.get_examples()
+        self.update_example_listboxes(self.ui.example_filename_before, self.ui.example_filename_after, examples)
+
+        if send_signal:
+            self.signal_update.emit()
+
+    def toggle_wordwrap(self):
+        """Toggles wordwrap in the script editing textbox.
+        """
+        if self.ui.file_naming_word_wrap.isChecked():
+            self.wordwrap = QtWidgets.QTextEdit.WidgetWidth
+        else:
+            self.wordwrap = QtWidgets.QTextEdit.NoWrap
+        self.ui.file_naming_format.setLineWrapMode(self.wordwrap)
+
+    def import_script(self):
+        """Import the current script from an external text file.
+        """
+        dialog_title = N_("Import Script File")
+        dialog_file_types = N_("All Files") + " (*);;" + N_("Picard Script Files") + " (*.pts)"
```

Comment:
Those strings are marked for translation (using `N_()`) but never actually translated (or I miss something).
Also `dialog_file_types` has the same value in `export_script()` so perhaps it could be made common.

Reply:
The selected file type is returned from the dialog and I had planned to use this on export to determine whether to write the file as plain text or JSON.  Since the JSON option isn't available right now, I'll make these a common property used in both import and export functions.